### PR TITLE
[DOC release] Update API docs for application initializer

### PR DESCRIPTION
**Pull request by swr**
Update application initializer example where store is accessed to pre load data. 'ember-data' must be loaded before we can access 'store', therefore 'ember-data' is put in an 'after' clause

http://emberjs.com/api/classes/Ember.Application.html#method_initializer

### CODE REVIEW

**Comment by rwjblue** [comment]
Forcing things to "materialize" eagerly (via `container.lookup` in an initializer) is generally not a good thing.  This was exactly why instance initializers were added.

The example using `this.container.lookup` should be changed to use an `Ember.Application.instanceInitializer` at which point the ordering does not matter (because all of initializers run before any of the instance initializers), would you mind taking a stab at that?

**Comment by swr** [comment]
Ahh. That does indeed sound better. Will modify later and update the pull request.

**Comment by rwjblue** [comment]
Awesome, thanks for working on it!

**Comment by swr** [comment]
I'm not seeing any refs to instanceInitializer in the release API docs, just the blog and changelog. Should this get broken out as a full doc section for the instanceInitializer method on application? AKA - put it in the methods list and such, and then make a reference to use cases for each in both initializers and instanceInitializers?

**Comment by rwjblue** [comment]
> Should this get broken out as a full doc section for the instanceInitializer method on application?

Yep, we should end up with a comment block just like the `initializer` one, for `instanceInitializer`.

**Comment by rwjblue** [comment]
@swr - ping

**Comment by swr** [comment]
Digging back in now.

**Comment by swr** [comment]
Broke it out into its own YUIdoc method and beefed up surrounding language. Let me know if you have any other feedback/changes.

File: packages/ember-application/lib/system/application.js [hunk 1]
```diff
@@ -980,6 +988,71 @@ Application.reopenClass({
     @param initializer {Object}
     @public
    */
+
+   /**
```
Comment (rwjblue) [inline packages/ember-application/lib/system/application.js]: Looks like this is causing the style checking tests to fail, I *think* it should be one space out-dented, but you can double check by running the tests locally.

**Comment by swr** [comment]
@rwjblue You were spot on on the source of the error. Thank you.

**Comment by rwjblue** [comment]
Awesome, thank you!

**Comment by swr** [comment]
@rwjblue Congrats on the 2.0 release! Been an exciting process to watch. Looks like this PR got lost in the shuffle to 2... should I restart it on current stable (2.0)? It is there on stable-1.13.

**Comment by rwjblue** [comment]
Yikes!  Sorry about that.  I'll get it pulled through there once I get all the branches fixed up....

**Comment by rwjblue** [comment]
OK, this is all pulled into release and canary (haven't branched for 2.1 beta just yet).  Thanks for pointing out that it wasn't there...

**Comment by swr** [comment]
No worries. Appreciate the effort. You have had your hands full.

**Pull request merged**